dojo/tools/rapplex/parser.py

```diff
@@ -0,0 +1,86 @@
+import json
+from datetime import datetime
+
+from dojo.models import Endpoint, Finding
+
+
+class RapplexParser:
+    """
+    Rapplex - Web Application Security Scanner
+    """
+    def get_scan_types(self):
+        return ["Rapplex Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "Rapplex Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "Import Rapplex JSON report."
+
+    def get_findings(self, filename, test):
+        data = json.load(filename)
+        findings = []
+        severities = ["Information", "Low", "Medium", "High", "Critical"]
+
+        for severity in severities:
+            current_severity = data.get("Severities", {}).get(severity)
+            if not current_severity:
+                continue
+
+            main_issue_groups = current_severity.get("IssueGroups", [])
+            for main_issue_group in main_issue_groups:
+                issues = main_issue_group.get("Issues", [])
+
+                for issue in issues:
+                    scan_id = data.get("ScanId", "")
+                    formatted_date = datetime.strptime(data.get("StartedDate", ""), "%d/%m/%Y %H:%M:%S").strftime("%Y-%m-%d")
+                    severity_level = current_severity.get("Name", "")
+                    title = issue.get("Title", "")
+                    url = issue.get("Url", "")
+                    req = issue.get("HttpRequest", "")
+                    res = issue.get("HttpResponse", "")
+                    vIndex = issue.get("vIndex", "")
+                    vuln_id = f"{scan_id}_{vIndex}"  # scanId and vIndex combined to create unique_id_from_tool
```

Comment:
In most tools, the scan ID is used to identify a single instance of a scan. Is this the case here as well? If so, deduplication will never be able to identify a match from scan to scan, as the unique ID from tool is actually TOO unique

Reply:
Yes, in Rapplex, the scan ID is determined specifically for each scan. unique_id_from_tool is not used now.